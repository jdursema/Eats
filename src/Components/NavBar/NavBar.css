header {
  background-color: #140802;
  opacity: .9; }
  header h1 {
    color: white;
    font-family: Courier New, Courier, monospace;
    font-weight: lighter;
    font-size: 80px;
    margin: 0; }
  header p {
    color: white; }
  header .title {
    display: flex;
    flex-direction: row;
    justify-content: space-between; }
  header .nav-bar {
    display: flex;
    justify-content: space-around; }
    header .nav-bar .nav {
      color: white;
      text-decoration: none;
      padding: 1%; }
