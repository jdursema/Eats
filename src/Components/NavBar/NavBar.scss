header{
  background-color: #0F2130;
  // opacity: .9;
  h1{
    color: white;
    font-family: 'IM Fell English SC', serif;
    font-weight: lighter;
    font-size: 80px;
    margin: 0;
  }
  p{
    color: white;
    font-family: 'Raleway', sans-serif;

  }
  .title{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
  .nav-bar{
    display: flex;
    justify-content: space-around;
    .nav{
      color: white;
      text-decoration: none;
      padding: 1%;
      font-family: 'Raleway', sans-serif;
    }
    
  }
  
}
