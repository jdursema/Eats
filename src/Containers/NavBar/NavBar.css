header {
  background-color: #0F2130; }
  header h1 {
    color: white;
    font-family: 'IM Fell English SC', serif;
    font-weight: lighter;
    font-size: 80px;
    margin: 0; }
  header p {
    color: white;
    font-family: 'Raleway', sans-serif; }
  header .title {
    display: flex;
    flex-direction: row;
    justify-content: space-between; }
  header .nav-bar {
    display: flex;
    justify-content: space-around; }
    header .nav-bar .nav {
      color: white;
      text-decoration: none;
      padding: 1%;
      font-family: 'Raleway', sans-serif;
      font-size: 20px; }

.log-out {
  border: white 2px solid;
  padding: 1%;
  width: 100px;
  color: white;
  text-decoration: none;
  margin: 3%;
  background-color: #0F2130;
  font-family: 'Raleway', sans-serif;
  font-size: 16px; }
  .log-out:hover {
    background-color: white;
    color: #0F2130; }
