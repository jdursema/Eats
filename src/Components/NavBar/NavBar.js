import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import './NavBar.css';

const NavBar = props => {
  return(
    <header>
      <div className='title'>
        <div className='left-nav'>
          <p>Hello, Guest!</p>
          <p>Your Location: <span>{props.location}</span></p>
        </div>
        <h1>Eats</h1>
        <div className='right-nav'>
          <NavLink to='/signup' className='nav'>SignUp</NavLink>
          <NavLink to='/login' className='nav'>Login</NavLink>
        </div>
      </div>
      <div className='nav-bar'>
        <NavLink to='/' className='nav'>Home</NavLink>
        <NavLink to='/suggestions' className='nav'>Suggestions</NavLink>
        <NavLink to='/favorites' className='nav'>Favorites</NavLink>
        <NavLink to='/recomendations' className='nav'>Recomendations</NavLink>
        <NavLink to='/location' className='nav'>Change Location</NavLink>
      </div>
    </header>
    
  );
};

const mapStateToProps = state => ({
  location: state.location
});

export default connect(mapStateToProps, null)(NavBar);