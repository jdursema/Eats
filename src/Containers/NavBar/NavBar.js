import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import './NavBar.css';
import { fetchLocation } from '../../Actions';
import PropTypes from 'prop-types';

class NavBar extends Component {

  async componentDidMount(){ 
    this.props.handleLocation();
  }

  render(){
    return (
      <header>
        <div className='title'>
          <div className='left-nav'>
            <p>Hello! {this.props.user.email}</p>
          </div>
          <h1>Eats</h1>
          <div className='right-nav'>
            <p>Your Location: {this.props.location.name}</p>
          </div>
        </div>
        <div className='nav-bar'>
          <NavLink to='/' className='nav'>
            Home
          </NavLink>
          <NavLink to='/suggestions' className='nav'>
            Places Near You
          </NavLink>
          <NavLink to='/favorites' className='nav'>
            Favorites
          </NavLink>
          <NavLink to='/recommendations' className='nav'>
            Recommendations
          </NavLink>
          <NavLink to='/location' className='nav'>
            Change Location
          </NavLink>
        </div>
      </header> 
    );
  }
}

const mapStateToProps = state => ({
  location: state.location,
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  handleLocation: () =>
    dispatch(fetchLocation())
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);


NavBar.propTypes = {
  handleLocation: PropTypes.func,
  location: PropTypes.object,
  user: PropTypes.object
};