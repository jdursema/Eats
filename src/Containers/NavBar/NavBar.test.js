import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { NavBar } from './NavBar';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('NavBar', () => {
  it('should match the snapshot', ()=> {
    const mockFunc = jest.fn();

    const renderedNavBar = shallow(<NavBar 
      handleLocation= { mockFunc }
      user={{}}
      location={{}}/>);

    expect(renderedNavBar).toMatchSnapshot();
  });

  it('should call handle location when the NavBar is mounted', () => {
    const mockFunc = jest.fn();

    shallow(<NavBar 
      handleLocation= { mockFunc }
      user={{}}
      location={{}}/>);

    expect(mockFunc.mock.calls.length).toEqual(1);

  })
});