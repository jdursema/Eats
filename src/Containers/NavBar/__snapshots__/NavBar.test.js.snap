// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    handleLocation={[Function]}
    location={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
>
          <div
                    className="left-nav"
          >
                    <p>
                              Hello! 
                    </p>
                    <p>
                              Your Location: 
                    </p>
          </div>
          <h1>
                    Eats
          </h1>
          <div
                    className="right-nav"
          >
                    <button
                              onClick={undefined}
                    >
                              LogOut
                    </button>
          </div>
</div>,
        <div
          className="nav-bar"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/suggestions"
          >
                    Places Near You
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/favorites"
          >
                    Favorites
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/recommendations"
          >
                    Recommendations
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/location"
          >
                    Change Location
          </NavLink>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="left-nav"
>
              <p>
                            Hello! 
              </p>
              <p>
                            Your Location: 
              </p>
</div>,
            <h1>
              Eats
</h1>,
            <div
              className="right-nav"
>
              <button
                            onClick={undefined}
              >
                            LogOut
              </button>
</div>,
          ],
          "className": "title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Hello! 
</p>,
                <p>
                  Your Location: 
</p>,
              ],
              "className": "left-nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Hello! ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hello! ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Your Location: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Your Location: ",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Eats",
            },
            "ref": null,
            "rendered": "Eats",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={undefined}
>
                LogOut
</button>,
              "className": "right-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LogOut",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "LogOut",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/suggestions"
>
              Places Near You
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/favorites"
>
              Favorites
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/recommendations"
>
              Recommendations
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/location"
>
              Change Location
</NavLink>,
          ],
          "className": "nav-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Places Near You",
              "className": "nav",
              "to": "/suggestions",
            },
            "ref": null,
            "rendered": "Places Near You",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Favorites",
              "className": "nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Recommendations",
              "className": "nav",
              "to": "/recommendations",
            },
            "ref": null,
            "rendered": "Recommendations",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Change Location",
              "className": "nav",
              "to": "/location",
            },
            "ref": null,
            "rendered": "Change Location",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
>
            <div
                        className="left-nav"
            >
                        <p>
                                    Hello! 
                        </p>
                        <p>
                                    Your Location: 
                        </p>
            </div>
            <h1>
                        Eats
            </h1>
            <div
                        className="right-nav"
            >
                        <button
                                    onClick={undefined}
                        >
                                    LogOut
                        </button>
            </div>
</div>,
          <div
            className="nav-bar"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/suggestions"
            >
                        Places Near You
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/favorites"
            >
                        Favorites
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/recommendations"
            >
                        Recommendations
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/location"
            >
                        Change Location
            </NavLink>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="left-nav"
>
                <p>
                                Hello! 
                </p>
                <p>
                                Your Location: 
                </p>
</div>,
              <h1>
                Eats
</h1>,
              <div
                className="right-nav"
>
                <button
                                onClick={undefined}
                >
                                LogOut
                </button>
</div>,
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Hello! 
</p>,
                  <p>
                    Your Location: 
</p>,
                ],
                "className": "left-nav",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Hello! ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Hello! ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Your Location: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Your Location: ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Eats",
              },
              "ref": null,
              "rendered": "Eats",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={undefined}
>
                  LogOut
</button>,
                "className": "right-nav",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LogOut",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "LogOut",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/suggestions"
>
                Places Near You
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/favorites"
>
                Favorites
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/recommendations"
>
                Recommendations
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/location"
>
                Change Location
</NavLink>,
            ],
            "className": "nav-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Places Near You",
                "className": "nav",
                "to": "/suggestions",
              },
              "ref": null,
              "rendered": "Places Near You",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Favorites",
                "className": "nav",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Recommendations",
                "className": "nav",
                "to": "/recommendations",
              },
              "ref": null,
              "rendered": "Recommendations",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Change Location",
                "className": "nav",
                "to": "/location",
              },
              "ref": null,
              "rendered": "Change Location",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
